### PR TITLE
golang 1.16 ARM post-submit

### PR DESCRIPTION
**Pull request by danbudris**
*Issue #, if available:*
https://github.com/aws/eks-distro-internal/issues/240

*Description of changes:*
Add a Go 1.16 ARM post-submit job. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by danbudris** [comment]
holding for a moment while I make some adjustments to the Golang Make file in build-tooling to properly account for the architecture in the upload path

File: templater/jobs/postsubmit/eks-distro-build-tooling/golang-1.16-ARM64-postsubmits.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+jobName: golang-1.16-tooling-postsubmit
+runIfChanged: projects/golang/go/1.16/.*
+runtimeImage: public.ecr.aws/amazonlinux/amazonlinux:2.0.20221004.0-arm64v8
```
Comment (danbudris) [inline templater/jobs/postsubmit/eks-distro-build-tooling/golang-1.16-ARM64-postsubmits.yaml:3]:
```suggestion
runtimeImage: public.ecr.aws/amazonlinux/amazonlinux:2
```

**Review by jaxesn — APPROVED** [review]

**Review by jaxesn — APPROVED** [review]

**Review by jaxesn — APPROVED** [review]

**Pull request merged**